homeassistant/components/derivative/sensor.py

```diff
@@ -165,6 +182,9 @@ def calc_derivative(event):
                 last_time, last_value = self._state_list[-1]
                 first_time, first_value = self._state_list[0]
 
+                if _has_overflowed(last_value, first_value):
+                    self._overflowed = True
```

Comment:
Why not something like:
```
self._overflowed = _has_overflowed(last_value, first_value)
```

Reply:
Yes, I tried that first. But that would lead to incorrect overflows.
In the end I tried this, combined with line 151: `self._overflowed = False`. And that worked as intended.

Reply:
About adding or not, the derivative function does not handle a byte-counter resetting to zero well. If we intend to extend the functionality of this integration, we should fix this as well. And the proposed solution does not interfere with the existing functionality. So why not add it?